plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/control/ArrayTableView.java

```diff
@@ -0,0 +1,63 @@
+package edu.wpi.first.shuffleboard.plugin.base.control;
+
+
+import javafx.beans.property.ReadOnlyObjectWrapper;
+import javafx.beans.property.ReadOnlyStringWrapper;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.TableCell;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.util.Callback;
+
+public class ArrayTableView<T> extends TableView<ArrayTableView.ArrayTableEntry<T>> {
+  private final TableColumn<ArrayTableEntry<T>, String> indexCol = new TableColumn<>("Index");
+  private final TableColumn<ArrayTableEntry<T>, T> valueCol = new TableColumn<>("Value");
+
+  private final ObservableList<ArrayTableEntry<T>> list = FXCollections.observableArrayList();
+
+  @SuppressWarnings("JavadocMethod")
+  public ArrayTableView() {
+    super();
+
+    indexCol.setCellValueFactory(p -> new ReadOnlyStringWrapper(String.valueOf(p.getValue().index)));
+    indexCol.setResizable(false);
+    indexCol.setPrefWidth(50);
+
+    valueCol.setCellValueFactory(p -> new ReadOnlyObjectWrapper<>(p.getValue().value));
+    valueCol.prefWidthProperty().bind(widthProperty().subtract(50).subtract(2));
+
+    //noinspection unchecked
+    getColumns().addAll(indexCol, valueCol);
+
+    setItems(list);
+  }
+
+  public void setValueCellFactory(
+      Callback<TableColumn<ArrayTableEntry<T>, T>, TableCell<ArrayTableEntry<T>, T>> callback
+  ) {
+    valueCol.setCellFactory(callback);
+  }
+
+  /**
+   * Convenience method to set the table's items.
+   */
+  public void setItems(T[] items) {
+    final var entries = new ArrayTableEntry[items.length];
+    for (int i = 0; i < items.length; i++) {
+      entries[i] = new ArrayTableEntry<>(i, items[i]);
+    }
+    //noinspection unchecked
+    list.setAll(entries);
```

Comment:
Using an arraylist would maintain type safety:

```suggestion
    final var entries = new ArrayList<ArrayTableEntry<T>>();
    for (int i = 0; i < items.length; i++) {
      entries.add(new ArrayTableEntry<>(i, items[i]));
    }

    list.setAll(entries);
```